x/tokenfactory/keeper/creators.go

```diff
@@ -0,0 +1,27 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k Keeper) addDenomFromCreator(ctx sdk.Context, creator, denom string) {
+	store := k.GetCreatorPrefixStore(ctx, creator)
+	store.Set([]byte(denom), []byte(denom))
+}
+
+func (k Keeper) getDenomsFromCreator(ctx sdk.Context, creator string) []string {
+	store := k.GetCreatorPrefixStore(ctx, creator)
+
+	iterator := store.Iterator(nil, nil)
+	defer iterator.Close()
+
+	denoms := []string{}
+	for ; iterator.Valid(); iterator.Next() {
+		denoms = append(denoms, string(iterator.Key()))
+	}
+	return denoms
+}
+
+func (k Keeper) GetAllDenomsIterator(ctx sdk.Context) sdk.Iterator {
+	return k.GetCreatorsPrefixStore(ctx).Iterator(nil, nil)
+}
```

Comment:
Very easy to reuse some code here:
```suggestion
func (k Keeper) getDenomsFromCreator(ctx sdk.Context, creator string) []string {
	iterator := k.GetAllDenomsIterator(ctx)
	defer iterator.Close()

	denoms := []string{}
	for ; iterator.Valid(); iterator.Next() {
		denoms = append(denoms, string(iterator.Key()))
	}
	return denoms
}

func (k Keeper) GetAllDenomsIterator(ctx sdk.Context) sdk.Iterator {
	return k.GetCreatorsPrefixStore(ctx).Iterator(nil, nil)
}

```